src/commands/upgrade.ts

```diff
@@ -176,13 +179,14 @@ export class UpgradeCommand extends Command {
                 }
 
                 const random = Math.random();
-                const success = random < condition.upgradeChance;
-                const downgrade = !success && Math.random() < 0.5;
+                const success = random < upgradeChance;
+                const downgrade = !success && Math.random() < downgradeChance;
                 if (success) {
                     currentCard.condition = condition.nextUpgrade;
```

Comment:
pretty sure pity should be reset on success

Reply:
typical pity systems would do that, but I don't think it would fit here

the main killer in the upgrade system is how much it can fluctuate, in 10% of cases your card downgrades 3+ times by going from good to poor and back on repeat for example
so resetting on success would be a bit pointless since it'd continue to allow those sequences to escalate costs anyway

at that point just reducing overall costs or reducing chance to downgrade in general would be more effective changes, which people would probably also be happy with